packages/common/src/stores/WatchHistoryController.ts

```diff
@@ -16,29 +12,19 @@ import { useWatchHistoryStore } from './WatchHistoryStore';
 @injectable()
 export default class WatchHistoryController {
   private readonly watchHistoryService: WatchHistoryService;
-  private readonly accountService?: AccountService;
 
-  constructor(@inject(INTEGRATION_TYPE) integrationType: IntegrationType, watchHistoryService: WatchHistoryService) {
+  constructor(watchHistoryService: WatchHistoryService) {
     this.watchHistoryService = watchHistoryService;
-    this.accountService = getNamedModule(AccountService, integrationType, false);
-  }
-
-  serializeWatchHistory = (watchHistory: WatchHistoryItem[]): SerializedWatchHistoryItem[] => {
-    return this.watchHistoryService.serializeWatchHistory(watchHistory);
-  };
 
-  private updateUserWatchHistory(watchHistory: WatchHistoryItem[]) {
-    const { user } = useAccountStore.getState();
+    // restore watch history when the user changes
+    useAccountStore.subscribe((state, previousState) => {
+      const isLoggedIn = !!state.user && !previousState.user;
+      const isLoggedOut = !state.user && !!previousState.user;
```

Comment:
It's more of an action than a state:
```suggestion
      const hasLoggedIn = !!state.user && !previousState.user;
      const hasLoggedOut = !state.user && !!previousState.user;
```

Reply:
removed subscriber for now